### ISSUE TITLE
When "su - user" (not root) Powerline left status doesn't show user@host

### ISSUE THREAD

**Issue by dlehenky** [romkatv/powerlevel10k#1255]
Title says it. "su user" _does_ display the correct status, as does "su -" for root, but not "su - user".
<img width="1241" alt="Untitled" src="https://user-images.githubusercontent.com/15184189/107457583-90072180-6b0f-11eb-82d4-2d9e5001f040.png">

Thanks!

**Comment by romkatv** [comment]
Yes, that is correct.

**Comment by dlehenky** [comment]
Are you saying that the current prompt behavior, as illustrated, is correct as it is? If so, why does 'su -' for root change the prompt? Really, it's important to know that you've 'su -' to an alternate unprivileged username, as it is for root, since both can result in different command behavior. And then there's the bit that just 'su username' _does_ change the prompt. Just trying to understand the reasoning here.

**Comment by romkatv** [comment]
> Are you saying that the current prompt behavior, as illustrated, is correct as it is?

The issue description is a statement. My comment was meant to express confirmation. "Your statement is correct. What you are saying is true."

If the behaviour you are describing doesn't match your expectations/desires, you need to describe the latter.

The standard template works quite well:

1. Describe what you do.
2. Describe what you expect.
3. Describe what you actually observe.

Points 1 and 2 are already clear. Point 3 is missing.

**Comment by dlehenky** [comment]
I believe you mean points 1 and 3 are clear, since I provided an actual illustration of what I do and the observed behavior. What I would expect (Point #2) is that the Powerline prompt, righthand element, would change for 'su - username' in the same manner it currently does for 'su - root', i.e. a consist outcome from like inputs.

**Comment by romkatv** [comment]
I'm not entirely sure I understand what you want (it would really help if you described it literally without allusions) but I'll try to describe what's possible.

In general, Powerlevel10k (or any other prompt) cannot display information based on what you did. It can only tell you what *is*. When you first log in as Alice and then switch to Bob, Powerlevel10k only knows that you are currently logged in as Bob. As such, it's indistinguishable from your being logged in as Bob to begin with.

If you always want to see what user you are logged in, see https://github.com/romkatv/powerlevel10k#how-do-i-add-username-andor-hostname-to-prompt.

If you want to see what user you are logged in *except when you are logged in as Alice*, do this:

```zsh
typeset -g POWERLEVEL9K_RIGHT_PROMPT_ELEMENTS=(...)  # note: no `context`
if [[ $USER != alice ]]; then
  POWERLEVEL9K_RIGHT_PROMPT_ELEMENTS+=(context)
fi
```

In other words, add `context` to your prompt conditionally.

Does this help?

**Comment by dlehenky** [comment]
Yes, thank you! I just commented out the context qualifier line (DEFUALT,SUDO) as per the tip in the file.

**Comment by romkatv** [comment]
> Yes, thank you! I just commented out the context qualifier line (DEFUALT,SUDO) as per the tip in the file.

👍 Thanks for posting this. It might help future users who face the same problem.

**Comment by FedeAbella** [comment]
Sorry to bring this back from the dead, but in case it helps anyone:
```bash
if [[ $(logname) != $USER ]] || [[ -n $SSH_CONNECTION ]]; then
    POWERLEVEL9K_RIGHT_PROMPT_ELEMENTS+=(context)
fi
```

works for only showing the context if you're logged in as another user (so with `su`) or via SSH.
